src/lib/ng-package/entry-point/analyse-sources.transform.ts

```diff
@@ -60,7 +60,7 @@ function analyseEntryPoint(graph: BuildGraph, entryPoint: EntryPointNode, entryP
 
   compilerHost.resolveModuleNames = (moduleNames: string[], containingFile: string) => {
     if (containingFile.endsWith('.d.ts')) {
-      return [].fill(undefined, 0, moduleNames.length);
+      return [].fill(undefined, 0, moduleNames.length + 1);
```

Comment:
I am a bit surprised this is looking at `containingFile`, not `resolvedModule.fileName`. Wouldn't that mean that the first reference into a `.d.ts` file is considered but not subsequent imports, instead of all `.d.ts` files?

Additionally, maybe using `modulesNames.map(() => undefined)` would be more readable.

Reply:
Agreed with the map. 

I don't follow the argument around `resolvedModule.fileName`, at this point we don't want to resolve modules if the import is non relative.